### ISSUE TITLE
LokiJS Adapter: useWebWorker: true, TypeError: LokiWebWorker is not a constructor

### ISSUE THREAD

**Issue by LibbyJax** [Nozbe/WatermelonDB#996]
I tried to enable the LokiJS Web Worker function and get an error.

"TypeError: LokiWebWorker is not a constructor".

React 17.0.2
@nozbe/watermelondb@0.22.0-2

This may just be my own lack of knowledge. Please forgive me if that's the case.

**Comment by radex** [comment]
@LibbyJax I think you didn't follow installation instructions to enable webpack to rewrite a magic import into a web worker. Follow those and see if it works. However, I generally would recommend against using web workers unless you're already using them and have a special need where things work better with web workers

**Comment by LibbyJax** [comment]
I will go back and revisit. The reason I was going to test web workers to see if it would alleviate the browser locking up on my first initial sync from a remote database with a large amount of records (Over 250,000+ records). Would you recommend another, better method? Currently, it runs the first sync and gets the large amount of changes down from the remote server and then it locks up the browser for about 30 minutes while, I assume, it is inserting the records. After that initial import, everything else has been working great.

**Comment by radex** [comment]
> Would you recommend another, better method? 

Yes - run profiler on your app to understand (not assume) why it takes it 30 minutes to insert 250k records, and then hopefully, fix the root cause.

What's your use case where you have that much work to do and (as long as the app isn't frozen), it's acceptable for initial sync to take half an hour?

Are you using Watermelon's built in `synchronize()` or are you implementing sync yourself?

**Comment by LibbyJax** [comment]
I will run profiler and get more information.

Currently, I have to use _unsafeBatchPerCollection: true or the the application runs out of memory rather quickly.

The use case where the data sync can happen in the background is that I can disable that portion of the app temporarily while that background sync completes. Since the big sync only happens on the first run, it is an acceptable workaround if I can keep the main rendering thread from freezing up.

I am using Watermelon's built-in synchronize with my own PHP back-end to handle the changes, updates, delete logic on the server side.

**Comment by xtealer** [comment]
> @LibbyJax I think you didn't follow installation instructions to enable webpack to rewrite a magic import into a web worker. Follow those and see if it works. However, I generally would recommend against using web workers unless you're already using them and have a special need where things work better with web workers

Just did the tutorial, and got this error.

<img width="1440" alt="Screen Shot 2021-08-28 at 6 57 38 AM" src="https://user-images.githubusercontent.com/45841958/131217419-61d8931e-35d0-4a6b-8d8d-34fa37018769.png">


ADAPTER:
```export default new LokiJSAdapter({
	schema,
	migrations,
	useWebWorker: false,
	useIncrementalIndexedDB: true,
	dbName: 'go_platform', // optional db name
	extraIncrementalIDBOptions: {
		onDidOverwrite: () => {
			// Called when this adapter is forced to overwrite contents of IndexedDB.
			// This happens if there's another open tab of the same app that's making changes.
			// Try to synchronize the app now, and if user is offline, alert them that if they close this
			// tab, some data may be lost
		},
		onversionchange: () => {
			// database was deleted in another browser tab (user logged out), so we must make sure we delete
			// it in this tab as well - usually best to just refresh the page
			const user = firebase.auth().currentUser;

			if (user) {
				window.location.reload();
			}
		},
	},
});
```

.BABELRC

```
{

	"plugins": [
		[
			"import",
			{
				"libraryName": "antd",
				"style": true
			},
			"transform-remove-console"
		],
		["@babel/plugin-proposal-decorators", { "legacy": true }],
		["@babel/plugin-proposal-class-properties", { "loose": true }],
		[
			"@babel/plugin-transform-runtime",
			{
				"helpers": true,
				"regenerator": true
			}
		]
	]
}

**Comment by mtourne** [comment]
> @LibbyJax I think you didn't follow installation instructions to enable webpack to rewrite a magic import into a web worker. Follow those and see if it works. However, I generally would recommend against using web workers unless you're already using them and have a special need where things work better with web workers

I've been having the same problem, and re-reading the docs many times and I can't find for the life of me the section of the docs with the magic import. Could you point out where this is described ?

**Comment by barbalex** [comment]
This error has popped up when trying to upgrade from vite v7 to vite v8. It is blocking us from upgrading vite
